src/sentry/api/event_search.py

```diff
@@ -1022,6 +1027,19 @@ def has_default(self, params):
         "args": [],
         "transform": "divide(countIf(notEquals(transaction_status, 0)), count())",
     },
+    "histogram": {
+        "name": "histogram",
+        "args": [
+            DurationColumnNoLookup("column"),
+            NumberRange("num_buckets", 1, 500),
+            NumberRange("bucket", 0, None),
+        ],
+        "column": [
+            "multiply",
+            [["floor", [["divide", [u"{column}", ArgValue("bucket")]]]], ArgValue("bucket")],
```

Comment:
Supporting this style of function in sentry and in snuba (for apdex) adds complexity in both applications. Would we have less overall complexity if this was implemented in snuba instead? I know that the AST. changes are blocked but we could use the apdex model for now.

Reply:
I don't think there will be less complexity, we will just move the complexity to snuba. I don't have as much experience with snuba so doing that will just slow development down. I think it's better to keep the complexity in sentry until the use cases fully crystallize.

Reply:
Ok. Supporting this function has required a good amount of additional code to our condition handling/resolution code. The snuba based approach wouldn't have been any more complicated than impact() is now, and could have re-used the 'hacks' we used to make impact() work.